### DISCUSSION TITLE
Streamdeck with Midi Plugin and Midi2LR

### DISCUSSION THREAD

**Discussion by RidniPhoto** [rsjaffe/MIDI2LR#736] [Ideas]
Hi!
Will it ever be possible to have a Streamdeck send out MiDi signals to Midi2LR through Midi Plugin for streamdeck?
I tried it with my Behringer but there is no way to move both a slider on the Streamdeck and on the Behringer...
I tried also to create a new virtual midi port, but both instruments cannot co-exist...

What do you think? Could it be done?

Matteo

**Comment by superpawko** [comment]
YOu can use it with loopMIDI program and MIDI plugin for streamdeck.  I created a virtual device in loopMIDI  and named it : streamDeck Port

in c:\Users\{your name}\AppData\Roaming\MIDI2LR\   is Controllers.xml  you have to set output to 0 , otherwise it will LOOP input. The file is created every time you close MIDI2LR window.

    <item devicename="streamDeck Port" systemid="\\?\root#media#0000#{6994ad04-93ef-11d0-a3cc-00a0c9223196}\tevmidi0"
          inputoutput="input" active="1"/>
    <item devicename="streamDeck Port" systemid="\\?\root#media#0000#{6994ad04-93ef-11d0-a3cc-00a0c9223196}\tevmidi0"
          inputoutput="output" active="0"/>

**Comment by RidniPhoto** [comment]
i appreciate your answer, however I’m on a mac…. Any ideas?

Regards

**Reply by superpawko** [reply]
https://dialogaudio.com/modulationprocessor/guides/virtual_midi/virtual_midi_setup.php
I use this one:
https://marketplace.elgato.com/product/midi-b068a591-1a69-48fe-9206-b2d24762228b

this one is only on mac i cant test it : 
https://marketplace.elgato.com/product/midi-button-c05a29fa-8080-4deb-96ac-8d8564dcdaa6

**Comment by RidniPhoto** [comment]
I created a virtual midi instrument on the Mac now,  (I tried also before but to no avail) but I'm still trying to figure out how to set up midi2lr...Il giorno 6 dic 2024, alle ore 18:48, superpawko ***@***.***> ha scritto:﻿
https://dialogaudio.com/modulationprocessor/guides/virtual_midi/virtual_midi_setup.php
I use this one:
https://marketplace.elgato.com/product/midi-b068a591-1a69-48fe-9206-b2d24762228b
this one is only on mac i cant test it :
https://marketplace.elgato.com/product/midi-button-c05a29fa-8080-4deb-96ac-8d8564dcdaa6

**Comment by superpawko** [comment]
Midi2LR detects all midi controllers, Just create virtual midi , use midi plugin on streamdeck to send a signal on selected KEY , channel on virtual device.
